sdk/servicebus/azure-servicebus/README.md

```diff
@@ -228,6 +228,11 @@ link will extend this timeout.
 a generator-style receive will run for before exiting if there are no messages.  Passing None (default) will wait forever, up until the 10 minute threshold if no other action is taken.
 - max_wait_time: Provided when calling receive() to fetch a batch of messages.  Dictates how long the receive() will wait for more messages before returning, similarly up to the aformentioned limits.
 
+### AutoLockRenew
+
+If for any reason auto-renewal has been interrupted or failed, this can be observed via the `auto_renew_error` property on the object being renewed.
+It would also manifest when trying to take action (such as completing a message) on the specified object.
+
```

Comment:
I guess the context here is that user uses AutoLockRenew but it fails to renew the object, so when user wants to do some op (e.g. complete), the op fails beyond user's expectation?

I'm thinking should we be more explicit when AutoLockRenew failed? -- currently we only assign the `auto_renew_error` value to the renewable object, and AutoLockRenew error can only be observed after user initiate an operation.

One simple way is escalating the logging level from "debug" to "warning" in AutoLockRenew when auto-renewal failure which improves visibility to some extent.

Thoughts?

Reply:
So this is actually one of the points being discussed in another customer thread currently (https://github.com/Azure/azure-sdk-for-python/issues/11611) and a callback would be the obvious approach to my eyes; it even meshes with the processor patterns in other languages that callback on-error, and would incrementally expose a user to the "I can be alerted if this fails" when registering the object to be renewed.  My primary motivation for this is that I figure a user isn't going to be able to programmatically respond to log messages, but your point may still be valid as well.

This is likely something we'd want to run past the architects but that's where my head is at right now.  My turn to ask "thoughts?"

Reply:
I mostly agree with the callback pattern if user wants explicit alert and to have control over the auto lock renew failure (e.g. removing the failed msg from current pending handling arry).

let's put this into the discussion.

Reply:
Created an issue to track this, https://github.com/Azure/azure-sdk-for-python/issues/11761